pandas/core/reshape/merge.py

```diff
@@ -952,6 +959,51 @@ def _validate_specification(self):
         if len(self.right_on) != len(self.left_on):
             raise ValueError("len(right_on) must equal len(left_on)")
 
+    def _validate(self):
+        # Get merging series:
```

Comment:
I tried that, but if I want to use `duplicated()`, it ends up taking just as much additional code (or more) to work with `self.right_join_keys` or `self.left_join_keys` -- I have to coerce them into DataFrames in a way that's amenable to their variable potential dimensions, then run duplicates. I don't think that's actually any better.